pkg/genericapiserver/default_storage_factory_builder.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package genericapiserver
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	"k8s.io/kubernetes/pkg/runtime"
+	etcdstorage "k8s.io/kubernetes/pkg/storage/etcd"
+	"k8s.io/kubernetes/pkg/util/config"
+)
+
+// defaultValue builds a DefaultStorageFactory.
+type DefaultStorageFactoryBuilder struct {
```

Comment:
Is the purpose of this to parse / enact runtime config? I tend to think that when the words "FactoryBuilder" appear in your source code it's a sign that it's time to reevaluate your life choices.

Can you at least change the comment to explain why building a default storage factory takes this much code?